### PR TITLE
[WIP] CF home page

### PR DESCRIPTION
**Pull request by blayhem**
image

### CODE REVIEW

File: src/components/expand/styles.scss [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    visibility: hidden;
+    opacity: 0;
+    transition: opacity 1s ease-in-out;
+    z-index: $z-index-expand-content;
+
+    &.visible {
```
Comment (sorodrigo) [inline src/components/expand/styles.scss]: please use correct notation

File: src/components/header/styles.scss [hunk 2]
```diff
@@ -1,18 +1,37 @@
 @import '~styles/settings';
 
 .c-header {
+  position: fixed;
```
Comment (sorodrigo) [inline src/components/header/styles.scss]: I'm not sure it needs to be fixed.

File: src/components/header/styles.scss [hunk 3]
```diff
@@ -1,18 +1,37 @@
...
+  z-index: $z-index-4;
 
-  .icon {
-    width: 20px;
-    max-height: 40px;
-    fill: $color-gray-1;
+  .menu-button {
+    height: 40px;
+    position: absolute;
+    font-size: 20px;
```
Comment (sorodrigo) [inline src/components/header/styles.scss]: let's talk about text styles, this is what I refer to when I talk about custom styles, better to have a closed API where classes cant define stuff such as `text-transform`, `font-size`, `font-weight` and `color`

File: src/components/icon/styles.scss [hunk 4]
```diff
@@ -1,8 +1,8 @@
 @import '~styles/settings';
 
 .c-icon {
-  width: 20px;
-  height: 20px;
+  width: 24px;
```
Comment (sorodrigo) [inline src/components/icon/styles.scss]: why not set `-medium` as default?

File: src/components/menu/styles.scss [hunk 5]
```diff
@@ -7,5 +7,40 @@
   width: 400px;
   top: 0;
   left: 0;
-  background: #343;
+  background: $color-gray-1;
+  color: $color-white;
+  font-size: 20px;
```
Comment (sorodrigo) [inline src/components/menu/styles.scss]: again custom styles

File: src/components/menu/styles.scss [hunk 6]
```diff
@@ -7,5 +7,40 @@
...
+  padding-top: 80px;
+
+  transform: translateX(-400px);
+  transition: transform 0.1s ease-in-out;
+
+  &.open {
```
Comment (sorodrigo) [inline src/components/menu/styles.scss]: use proper notation for modifiers

File: src/components/menu/styles.scss [hunk 7]
```diff
@@ -7,5 +7,40 @@
...
+    transform: translateX(0);
+  }
+
+  .close-button {
+    position: absolute;
+    top: 16px;
+    right: 16px;
+    text-transform: uppercase;
```
Comment (sorodrigo) [inline src/components/menu/styles.scss]: custom styles

File: src/components/expand/styles.scss [hunk 8]
```diff
@@ -0,0 +1,65 @@
+@import '~styles/settings';
+
+$z-index-blender: 2;
+$z-index-expand-content: 3;
```
Comment (sorodrigo) [inline src/components/expand/styles.scss]: dont use local `z-index` variables. They should be global and generic

File: src/components/expand/styles.scss [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+      visibility: visible;
+      opacity: 1;
+    }
+  }
+
+  .toggle {
+    right: calc(50% - #{$button-width / 2});
+    bottom: 1rem;
+    background: $color-gray-2;
+    color: $color-white;
```
Comment (sorodrigo) [inline src/components/expand/styles.scss]: custom styles

File: src/components/header/styles.scss [hunk 10]
```diff
@@ -1,18 +1,37 @@
...
+    text-transform: uppercase;
+    display: flex;
+    align-items: center;
+    justify-content: center;
     cursor: pointer;
+
+    .menu-icon {
+      margin-right: 16px;
+      position: relative;
+      bottom: 2px;
```
Comment (sorodrigo) [inline src/components/header/styles.scss]: why do you need this? If the reason is to better center the icon, use a `transform`.

File: src/components/header/styles.scss [hunk 11]
```diff
@@ -1,18 +1,37 @@
...
+    }
+  }
+
+  span {
+    width: 100%;
+    text-align: center;
```
Comment (sorodrigo) [inline src/components/header/styles.scss]: custom styles

File: src/pages/home/component.jsx [hunk 12]
```diff
@@ -1,10 +1,65 @@
...
+    }
+  ];
+  const [sectionIndex, setSection] = useState(0);
+  const [transition, setTransition] = useState('fade-in');
+  const section = sections[sectionIndex];
+
+  const changeSection = next => {
+    setTransition('fade-out');
+    setTimeout(() => setSection(next), 200);
+    setTimeout(() => setTransition('fade-in'), 300);
```
Comment (sorodrigo) [inline src/pages/home/component.jsx]: I dont love this, why not paint both and toggle the opacity of the active one and the inactive one. Or if that's too complicated, why not use `react-spring`? If you want, we can talk about this offline.

Comment (sorodrigo) [reply src/pages/home/component.jsx]: Also you're not clearing these timeouts

File: src/pages/home/styles.scss [hunk 13]
```diff
@@ -1,5 +1,73 @@
...
 
+    width: 100vw;
+    height: 100vh;
+    display: flex;
+    flex-direction: column;
+    align-content: center;
+    justify-content: center;
+    transition: opacity 0.2s ease-in-out;
+
+    &.fade-out {
```
Comment (sorodrigo) [inline src/pages/home/styles.scss]: use proper notation

File: src/pages/home/styles.scss [hunk 14]
```diff
@@ -1,5 +1,73 @@
 @import '~styles/settings';
 
 .c-home {
+  .section {
+    font-family: $font-family-2;
```
Comment (sorodrigo) [inline src/pages/home/styles.scss]: custom styles

File: src/pages/home/styles.scss [hunk 15]
```diff
@@ -1,5 +1,73 @@
...
+    color: $color-gray-1;
+    padding: 0 0 160px 160px;
 
+    width: 100vw;
+    height: 100vh;
+    display: flex;
```
Comment (sorodrigo) [inline src/pages/home/styles.scss]: do we need this flex?

File: src/pages/home/styles.scss [hunk 16]
```diff
@@ -1,5 +1,73 @@
...
+
+    &.fade-out {
+      opacity: 0;
+    }
+
+    &.fade-in {
+      opacity: 1;
+    }
+
+    h1 {
```
Comment (sorodrigo) [inline src/pages/home/styles.scss]: super custom styles

File: src/pages/home/styles.scss [hunk 17]
```diff
@@ -1,5 +1,73 @@
...
+      font-weight: 800;
+      font-size: 64px;
+      line-height: 84px;
+      max-width: 600px;
+      margin-bottom: 44px;
+    }
+
+    p{
```
Comment (sorodrigo) [inline src/pages/home/styles.scss]: super custom styles

File: src/pages/home/styles.scss [hunk 18]
```diff
@@ -1,5 +1,73 @@
...
+}
+
+.c-choose-country {
+  width: 100%;
+  height: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  button {
```
Comment (sorodrigo) [inline src/pages/home/styles.scss]: use class names instead of elements

**Comment by sorodrigo** [comment]
I've seen that you're targeting the `master` branch. I know I've asked this before, please create a develop branch and set it up as the default branch 🙏 .

File: src/components/expand/styles.scss [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+    position: absolute;
+    left: calc(50% - #{$button-height / 2});
+    bottom: 1rem;
+
+    height: $button-height;
+    width: $button-height;
+    margin: auto;
+    border-radius: 50%;
+    background: white;
+    transition: transform 1s ease-in-out;
```
Comment (sorodrigo) [inline src/components/expand/styles.scss]:
```suggestion
    transition: transform 0.25s cubic-bezier(.445, .05, .55, .95); // $ease-in-out-sine
```

File: src/components/expand/styles.scss (hunk 1 above)
Comment (sorodrigo) [reply src/components/expand/styles.scss]: I think you meant this comment. Lets use either bem or the vizzuality's notation (dont know the name), in which case modifiers should start with a dash: `-visible`

File: src/pages/home/styles.scss (hunk 15 above)
Comment (blayhem) [reply src/pages/home/styles.scss]: to center content easily, maybe

**Pull request merged**